tests/devapps/NetCoreTestApp/Program.cs

```diff
@@ -241,6 +244,33 @@ 1. IWA
 
                             break;
 
+                        case 11: // AcquireTokenForClient with multiple threads
+                            Console.Write("Enter number of threads to start (default 10): ");
+                            int totalThreads = int.TryParse(Console.ReadLine(), out totalThreads) ? totalThreads : 10;
+                            Console.Write("Enter run duration in seconds (default 10): ");
+                            int durationInSeconds = int.TryParse(Console.ReadLine(), out durationInSeconds) ? durationInSeconds : 10;
+
+                            var acquireTokenBuilder = CreateCca().AcquireTokenForClient(GraphAppScope);
+
+                            var threads = new List<Thread>();
+                            for (int i = 0; i < totalThreads; i++)
+                            {
+                                var thread = new Thread(new ThreadStart(new ThreadWork(i, acquireTokenBuilder, durationInSeconds).Run));
```

Comment:
nit: Threads are so 2015... Task.WhenAll() is the way you can do the same with Tasks. See https://github.com/AzureAD/microsoft-authentication-extensions-for-dotnet/blob/e7f579d3f9db9acc2e9225356ab7a72b956949ec/sample/ManualTestApp/Program.cs#L124

Reply:
Tasks can run on the same thread though right? With threads, it's guaranteed separate.